devdoc/job_queue.md

```diff
@@ -0,0 +1,59 @@
+# `job_queue` requirements
+
+`job_queue` is a module that allows jobs to be executed asynchronously. It queues jobs when there are no requests and queues requests when there are no jobs. When a job arrives and there are queued requests, it executes the least recently queued request with the job. When a request arrives and there are queued jobs, it executes the request with the least recently queued job.
```

Comment:
>It queues jobs when there are no requests [](http://example.com/codeflow?start=72&length=41)

What does this mean? What if there are requests, what happens to jobs?

Reply:
It means that if jobs are pushed when no requests have been submitted, they will be stored in a queue until a session submits requests to consume them.

If a job is submitted when there are pending requests, the `JOB_REQUEST_FUNCTION` for the least recently submitted job for each session is executed with the submitted job as parameter.